### PR TITLE
fix segfault when calling FunctionField on empty vector

### PR DESCRIPTION
**Pull request by rfourquet**
I didn't investigate enough to be certain that this fix is correct, maybe @raulepure can confirm.

### CODE REVIEW

**Comment by raulepure** [comment]
I tried the fix on some small examples and it seems to work.  

While looking at your fix, I have the following suggestion: 
We should avoid that someone passes  a vector containing empty strings (i.e. K, (a,b) = FunctionField(QQ, ["",""])).
This seems to work, but I'm not too confident with what is send to Singular.

**Comment by rfourquet** [comment]
> We should avoid that someone passes a vector containing empty strings (i.e. K, (a,b) = FunctionField(QQ, ["",""])).

Sounds like a good idea to prevent empty strings, I will add that.

**Comment by raulepure** [comment]
Thank you! 

I noticed two further things this morning: 
- the problem with the empty strings appears in every PolyRing constructor and will have to be fixed there too in other pull requests.  
- We may have to prevent users from passing multiple times the same string, like K, (a,b) = FunctionField(QQ, ["a", "a"])

In principle the last point cause no harm on the Singular side, but it is definitively not desired and Singular does not allow it as user input.
It could be fixed by something like 

unique(S) != S && throw(ArgumentError("strings in array must be pairwise different"))

**Comment by rfourquet** [comment]
> We may have to prevent users from passing multiple times the same string, like K, (a,b) = FunctionField(QQ, ["a", "a"])

I also wondered about that but saw that it seemed to work as expected. But while I'm at it, if the singular interpreted disallows it, I will fix it here.

File: src/number/n_transExt.jl [hunk 1]
```diff
@@ -369,15 +369,16 @@ end
...
 with transcendence basis stored in the array $S$.
 """
-function FunctionField(F::Singular.Field, S::Array{String, 1}; cached::Bool=true)
+function FunctionField(F::Singular.Field, S::Vector{String}; cached::Bool=true)
    typeof(F) == N_GField && error("Finite field extensions of Fp not supported.")
-   l = length(S)
-   isempty(l) && throw(DomainError())
+   isempty(S) && throw(ArgumentError("array must be non-empty"))
+   any(isempty, S) && throw(ArgumentError("strings in array must be non-empty"))
+   allunique(S) && throw(ArgumentError("strings in array must be pairwise different"))
```
Comment (raulepure) [inline src/number/n_transExt.jl:381]: I think you mean !allunique

Comment (raulepure) [reply src/number/n_transExt.jl:381]: And then the tests should all pass

Comment (rfourquet) [reply src/number/n_transExt.jl]: Yes thank you I also just noticed because of failing tests!

Comment (raulepure) [reply src/number/n_transExt.jl]: Now all tests passed. I will merge.

**Pull request merged**